traits/tests/test_ctraits.py

```diff
@@ -188,3 +188,8 @@ def test_comparison_mode_enum(self):
 
         self.assertIsInstance(trait.comparison_mode, ComparisonMode)
         self.assertEqual(trait.comparison_mode, ComparisonMode.equality)
+
+    def test_assign_post_setattr_none(self):
+        trait = CTrait(TraitKind.trait)
+
+        trait.post_setattr = None
```

Comment:
I think this test can and should be expanded.

What you're testing here is simply that assignment of `None` doesn't raise. That's a weak test, that allows for all sorts of other failure modes. It's fine as a regression test, but given that this feature is a bit more than simply a bugfix, we should do better testing.

One way to think of this: the assignment `trait.post_setattr = None` should have some observable effect. (If not, there's little point supporting it.) Ideally, a unit test should test that observable effect. In this case, setting `None` means that the function that would previously have been called should no longer be called. So I'd suggest a test that:

- sets a non-None `post_setattr` callable
- verifies that that callable is called on trait set
- sets `post_setattr` back to `None`
- verifies that the previous callable is no longer called on trait set

As another rule of thumb: if you find yourself writing a unit test that doesn't have any `self.assert*` calls in it, there's probably scope to improve the test. In this case, an obvious and easy improvement is to add a line asserting that after the assignment of `None`, `trait.post_setattr` is retrieved as `None`.

Reply:
Made those changes. Also discovered a bug in my implementation and fixed it. Thank you for the advice!

Reply:
New tests look good; thank you!